timesync/converter.go

```diff
@@ -0,0 +1,27 @@
+package timesync
+
+import (
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"time"
+)
+
+type LayerConv struct {
+	duration time.Duration // the layer duration
+	genesis  time.Time     // the genesis time
+}
+
+// TimeToLayer returns the layer of the provided time
+func (lc LayerConv) TimeToLayer(t time.Time) types.LayerID {
+	if t.Before(lc.genesis) { // the genesis is in the future
+		return 0
+	}
+	return types.LayerID((t.Sub(lc.genesis) / lc.duration) + 1)
```

Comment:
you dont check that `duration` > 0

Reply:
Fixed it in NewClock. Zaro duration is a bad configuration and will result in a panic.